src/ModelParser/VisibilityAwarePropertyAccessGuesser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Liip\MetadataParser\ModelParser;
+
+use Liip\MetadataParser\Exception\ParseException;
+use Liip\MetadataParser\Metadata\PropertyAccessor;
+use Liip\MetadataParser\ModelParser\RawMetadata\PropertyVariationMetadata;
+use Liip\MetadataParser\ModelParser\RawMetadata\RawClassMetadata;
+use ReflectionClass;
+use ReflectionException;
+
+class VisibilityAwarePropertyAccessGuesser implements ModelParserInterface
+{
+    public function parse(RawClassMetadata $classMetadata): void
+    {
+        try {
+            $reflClass = new ReflectionClass($classMetadata->getClassName());
+        } catch (ReflectionException $e) {
+            throw ParseException::classNotFound($classMetadata->getClassName(), $e);
+        }
+
+        foreach ($classMetadata->getPropertyCollections() as $propertyCollection) {
+            foreach ($propertyCollection->getVariations() as $variation) {
+                $name = $variation->getName();
+
+                if (!$reflClass->hasProperty($name) || $reflClass->getProperty($name)->isPublic()) {
+                    continue;
+                }
+
+                $currentAccessor = $variation->getAccessor();
+
+                if (!($currentAccessor->getSetterMethod() && $currentAccessor->hasSetterMethod())) {
+                    $variation->setAccessor(new PropertyAccessor(
+                        $currentAccessor->getGetterMethod() ?: $this->guessGetter($reflClass, $variation),
+                        $currentAccessor->getSetterMethod() ?: $this->guessSetter($reflClass, $variation),
+                    ));
+                }
+            }
+        }
+    }
+    private function guessGetter(ReflectionClass $reflClass, PropertyVariationMetadata $variation): ?string
+    {
+        foreach (['get', 'is', ''] as $prefix) {
+            $method = "$prefix{$variation->getName()}";
+
+            if (!$reflClass->hasMethod($method)) continue;
+
+            $reflMethod = $reflClass->getMethod($method);
+
+            if ($reflMethod->isPublic() && ($reflMethod->getNumberOfRequiredParameters() === 0)) {
+                return $method;
+            }
+        }
+
+        return null;
+    }
+
+    private function guessSetter(ReflectionClass $reflClass, PropertyVariationMetadata $variation): ?string
+    {
+        foreach (['set', ''] as $prefix) {
+            $method = "$prefix{$variation->getName()}";
+
+            if (!$reflClass->hasMethod($method)) continue;
+
+            $reflMethod = $reflClass->getMethod($method);
+
+            if ($reflMethod->isPublic() && ($reflMethod->getNumberOfParameters() >= 1)) {
```

Comment:
what if there is more than 1 parameter expected, will the generated code work (if the additional parameters are required)?